src/index.js

```diff
@@ -480,7 +480,14 @@ class Offline {
                   return errorResponse(reply);
                 }
               }
-              else {
+              else if (request.auth && request.auth.credentials && 'usageIdentifierKey' in request.auth.credentials) {
+                const usageIdentifierKey = request.auth.credentials['usageIdentifierKey'];
```

Comment:
I did not merge because of the lint error here: `request.auth.credentials['usageIdentifierKey']` should be `request.auth.credentials.usageIdentifierKey`